src/BenchmarksApps/HttpClientBenchmarks/Clients/HttpClient/Program.cs

```diff
@@ -0,0 +1,268 @@
+﻿using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Diagnostics;
+using System.Net.Security;
+using Microsoft.Crank.EventSources;
+
+namespace HttpClientBenchmarks;
+
+class Program
+{
+    private static ClientOptions s_options = null!;
+    private static List<HttpClient> s_httpClients = new();
+    private static Metrics s_metrics = new();
+    private static bool s_isWarmup;
+    private static bool s_isRunning;
```

Comment:
Probably not a big issue, but you may want to consider some synchronization around these variables, at least by making them [`volatile`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/volatile). (Even better if you can avoid them, by making warmup a separate step.)

Reply:
They were two separate steps before, but I merged them together as per @geoffkizer's comment https://github.com/aspnet/Benchmarks/pull/1706#discussion_r746996563

Some time ago, I've discussed `volatile` and `Interlocked` and other synchronization for a flag set by one thread and read by another thread with @stephentoub -- and the bottom line was that it's not really going to help with anything.
1) `volatile` would help only with ordered assignments -- if there are two flags and I depend on their order of assignment in the other thread(s).
2) `Interlocked` is only needed if there could be different threads assigning the same variable.
So neither of them would help me see the assigned variable "faster" in the other thread. It's a boolean, so there shouldn't be a "broken" read. And I don't care that much to see the value as soon as possible - I measure the time separately inside the tasks, not globally outside them, so as soon as the value change is seen in a reasonable time and the loop is stopped, I should be fine.